src/main/rx/expresslrs.c

```diff
@@ -1059,65 +1057,134 @@ static void enterBindingMode(void)
 
 static uint32_t isrTimeStampUs;
 
+typedef enum {
+    RX_ELRS_STATE_GETIRQ,
+    RX_ELRS_STATE_ISR,
+    RX_ELRS_STATE_PROCESS,
+    RX_ELRS_STATE_FHSS,
+    RX_ELRS_STATE_TELEM,
+    RX_ELRS_STATE_CONN,
+    RX_ELRS_STATE_COUNT
+} expressLrsState_e;
+
+static expressLrsState_e expressLrsState = RX_ELRS_STATE_GETIRQ;
+
+// Un-comment define to split state into a separate phase
+//#define USE_ELRS_STATE_ISR
+//#define USE_ELRS_STATE_PROCESS
+//#define USE_ELRS_STATE_FHSS
+//#define USE_ELRS_STATE_TELEM
+
 rx_spi_received_e expressLrsDataReceived(uint8_t *payload)
 {
+    static uint8_t irqReason;
     rx_spi_received_e result = RX_SPI_RECEIVED_NONE;
 
-    if (!receiver.started && (systemState & SYSTEM_STATE_READY)) {
-        receiver.started = true;
-        startReceiving(); // delay receiving after initialization to ensure a clean connect
-    }
+    pinioSet(1, 1);
+    switch(expressLrsState) {
```

Comment:
I'm not sure here. It may be better to set `expressLrsState` even when processing multiple steps in single call and use `again: switch () {` and `goto again;` to process next state. Or wrap switch ( `while (true) switch () { }; break; }') and use `continue` to repeat `switch`.
There will be small speed/size penalty, but code readability may improve considerably.

Reply:
This is only temporary as I include/exclude states. I'll clean it up when I'm done.